### PR TITLE
Real time eth price scores

### PR DESCRIPTION
**Pull request by 0xZenus**

### CODE REVIEW

File: src/entities/communitySbt.ts [hunk 1]
```diff
@@ -635,39 +646,34 @@ class SBT {
    * the awarded time as end of season and claimed time
    * as eithr zero if not claimed or subgrap's minted timestamp
    */
-  async constructTopBadge(
+  static async constructTopBadge(
```
Comment (ioanaroceanu2000) [inline src/entities/communitySbt.ts:649]: what does static mean in this context?

File: src/entities/communitySbt.ts [hunk 2]
```diff
@@ -553,80 +539,105 @@ class SBT {
...
 
-    if (!this.ethPrice) {
-      this.ethPrice = await geckoEthToUsd(this.coingeckoKey);
+    return {
+      topTraderBadge,
+      topLpBadge,
+    };
+  }
+
+  public async getRanking(seasonId: number): Promise<{
```
Comment (ioanaroceanu2000) [inline src/entities/communitySbt.ts:602]: saw you removed some params of this function, including ignored wallets list. How do we avoid including voltz addresses in the ranking?

File: src/entities/communitySbt.ts [hunk 3]
```diff
@@ -553,80 +539,105 @@ class SBT {
...
-      }
-      const entry = rankResult[rank];
-      if (entry.address === userId) {
-        const badge = await this.constructTopBadge(
+    const traderPosition = rankResult.traderRankResults
+      .slice(0, 5)
+      .find((rank) => rank.address.toLowerCase() === userId.toLowerCase());
+
+    const topTraderBadge = !isUndefined(traderPosition)
+      ? await SBT.constructTopBadge(
```
Comment (ioanaroceanu2000) [inline src/entities/communitySbt.ts:573]: why SBT.f() instead of this.f()?

File: src/utils/communitySbt/getScores.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  lpScores: Record<string, number>;
+  traderScores: Record<string, number>;
+};
+
+/**
+ * @dev Query the Main subgraph and retrieve season's liquidity
+ * or trading score of all users based on time weighted liquidity.
+ * Score is based on swaps for Traders and mints/burns for LPs.
+ */
+export async function getScores({
```
Comment (ioanaroceanu2000) [inline src/utils/communitySbt/getScores.ts:19]: for leaderboard, we are only interested in trader scores. do you think also computing LP scores can affect performance?

File: src/entities/communitySbt.ts [hunk 5]
```diff
@@ -855,7 +837,7 @@ class SBT {
       this.currentBadgesSubgraphUrl,
       this.nextBadgesSubgraphUrl,
     );
-    const subgraphClaimedBadges = await this.claimedBadgesInSubgraph(
+    const subgraphClaimedBadges = await SBT.claimedBadgesInSubgraph(
```
Comment (ioanaroceanu2000) [inline src/entities/communitySbt.ts:840]: again, why SBT and not this?

File: src/entities/communitySbt.ts (hunk 1 above)
Comment (0xZenus) [reply src/entities/communitySbt.ts:649]: static is used when the function doesn't depend on the object itself (i.e. it doesn't use any fields of ``this``)

File: src/entities/communitySbt.ts (hunk 3 above)
Comment (0xZenus) [reply src/entities/communitySbt.ts:573]: because the function is ``static`` at this point

File: src/entities/communitySbt.ts (hunk 5 above)
Comment (0xZenus) [reply src/entities/communitySbt.ts:840]: because the function is ``static`` at this point

File: src/utils/communitySbt/getScores.ts (hunk 4 above)
Comment (0xZenus) [reply src/utils/communitySbt/getScores.ts:19]: That's right, but all data is fetched at once so the difference is quite insignificant.

File: src/entities/communitySbt.ts (hunk 2 above)
Comment (0xZenus) [reply src/entities/communitySbt.ts:602]: that variable wasn't used inside the function. What is used is the internal field ``this.ignoredWalletIDs``

**Review by ioanaroceanu2000 — APPROVED** [review]

**Pull request merged**